src/service/video-manager-impl.js

```diff
@@ -894,13 +945,235 @@ class VideoEntry {
     // Update docking state
     if (this.scrollMap_(DOCK_SCALE, 1) == DOCK_SCALE) {
       this.dockState_ = DockStates.DOCKED;
+      this.initializeDragging_();
+      this.drag_();
     } else {
+      this.finishDragging_();
       this.dockState_ = DockStates.DOCKING;
     }
 
     // TODO(@wassim) Make minimized video draggable
   }
 
+
+  initializeDragging_() {
+    if (this.dragListenerInstalled_) {
+      return;
+    }
+    const minimizedRect = this.internalElement_./*OK*/getBoundingClientRect();
+    this.dragCoordinates_.initial.x = minimizedRect.left;
+    this.dragCoordinates_.initial.y = minimizedRect.top;
+    this.dragCoordinates_.position.x = minimizedRect.left;
+    this.dragCoordinates_.position.y = minimizedRect.top;
+    this.dragCoordinates_.previous.x = minimizedRect.left;
+    this.dragCoordinates_.previous.y = minimizedRect.top;
+
+    this.draggingMask_ = this.createDraggingMask_();
+
+    // Desktop listeners
+    listen(this.draggingMask_, 'mousedown', e => {
+      e.preventDefault();
+      this.isTouched_ = true;
+      this.isDragging_ = false;
+      this.mouse_(e, true);
+    });
+    listen(this.ampdoc_.win.document, 'mouseup', () => {
+      this.isTouched_ = false;
+      this.isDragging_ = false;
+    });
+    listen(this.ampdoc_.win.document, 'mousemove', e => {
+      this.isDragging_ = this.isTouched_;
+      if (this.isDragging_) {
+        e.preventDefault();
+      }
+      this.mouse_(e);
+    });
+    // Touch listeners
+    listen(this.draggingMask_, 'touchstart', e => {
+      e.preventDefault();
+      this.isTouched_ = true;
+      this.isDragging_ = false;
+      this.mouse_(e, true);
+    });
+    listen(this.ampdoc_.win.document, 'touchend', () => {
+      this.isTouched_ = false;
+      this.isDragging_ = false;
+    });
+    listen(this.ampdoc_.win.document, 'touchmove', e => {
+      this.isDragging_ = this.isTouched_;
+      if (this.isDragging_) {
+        e.preventDefault();
+      }
+      this.mouse_(e);
+    });
+    this.dragListenerInstalled_ = true;
+  }
+
+  /**
+   * Handles the dragging, dropping and snapping to corners.
+   * Ran once every animation frame
+   * @private
+   */
+  drag_() {
+    // Stop the loop if the video is no longer in a draggable state
+    if (!this.loaded_
+      || !this.internalElement_
+      || this.dockPosition_ == DockPositions.INLINE
+      || this.dockVisibleHeight_ != 0
+      || !this.internalElement_.classList.contains(DOCK_CLASS)
+      || this.dockState_ != DockStates.DOCKED) {
+      return;
+    }
+
+    const minimizedRect = this.internalElement_./*OK*/getBoundingClientRect();
```

Comment:
let's keep `drag_` a mutate-only function. Does it need to read this every time? can it by cached by `initializeDragging_`?